pennylane/circuit_graph.py

```diff
@@ -116,6 +117,58 @@ def __init__(self, ops, variable_deps):
                 # Create an edge between this and the previous operator
                 self._graph.add_edge(wire[i - 1], wire[i])
 
+    def create_hash_argument(self):
+        """Create a unique hash based on the operations and observables in the circuit graph and the index of the variables
+        used by them.
+        """
+        key_to_be_hashed = ''
+
+        delimiter = '!'
+
+        for op in self.operations_in_order:
+            key_to_be_hashed += op.name
+
+            for param in op.params:
+                if isinstance(param, Variable):
+                    key_to_be_hashed += delimiter
+                    key_to_be_hashed += str(param.idx)
+                    key_to_be_hashed += delimiter
+
+                else:
+                    key_to_be_hashed += delimiter
+                    key_to_be_hashed += str(param)
+                    key_to_be_hashed += delimiter
+
+            key_to_be_hashed += str(op.wires)
+
+        # Adding a distinct separating string that could not occur by any combination of the
+        # name of the operation and wires
+        key_to_be_hashed += '|||'
+
+        for obs in self.observables_in_order:
+            key_to_be_hashed += str(obs.name)
+            for param in obs.params:
+                if isinstance(param, Variable):
+                    key_to_be_hashed += delimiter
+                    key_to_be_hashed += str(param.idx)
+                    key_to_be_hashed += delimiter
+
+                else:
+                    key_to_be_hashed += delimiter
+                    key_to_be_hashed += str(param)
+                    key_to_be_hashed += delimiter
+
+
+            key_to_be_hashed += str(obs.wires)
+
+        return key_to_be_hashed
+
+    @property
+    def circuit_hash(self):
```

Comment:
Since the class is already `CircuitGraph`, might be nicer just to call this property `hash`, i.e., `CircuitGraph.hash` (avoids the repeated use of circuit)

Reply:
True! I was hesitant to make it that since ``hash`` is reserved as a keyword, but I do agree with your point, I'll change it so

Reply:
Should be fine as long as we never create a `hash` function 😆